### PR TITLE
Issue 1656 - DynamicForm and Form components + tests update

### PR DESCRIPTION
**Pull request by alekpentchev**
# Description

DynamicForm and Form components were rewritten from class-based components to functional components. Tests have been updated for both components.

## Corresponding Issue

https://github.com/ifmeorg/ifme/issues/1656
# Screenshots

Screenshot 2020-03-05 at 17 15 15

---

Reviewing this pull request? Check out our [Code Review Practices](https://github.com/ifmeorg/ifme/wiki/Code-Review-Practices) guide if you haven't already!

### CODE REVIEW

**Comment by alekpentchev** [comment]
circleci test is failing probably because of [this issue in eslint about props validation](https://github.com/yannickcr/eslint-plugin-react/issues/1280)

File: INSTALLATION.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
+# Installation
```
Comment (julianguyen) [inline INSTALLATION.md]: We don't need our installation doc to be here as it's in our [Wiki](https://github.com/ifmeorg/ifme/wiki/Installation). If you want to make changes, please update the existing doc :)

File: client/app/components/Form/DynamicForm.jsx [hunk 2]
```diff
@@ -8,63 +9,69 @@ import { REQUIRES_DEFAULT } from '../Input/InputDefault';
...
 
+type FormProps = {
+  action: string,
+  inputs: MyInputProps[],
+};
+
 type Errors = { [string]: boolean } | {};
 
 export type Props = {
+  // Somehow Flow does not detect nameValue being used in a function outside the component
```
Comment (julianguyen) [inline client/app/components/Form/DynamicForm.jsx]: Oh no, that's strange :O

File: client/app/components/Form/DynamicForm.jsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 // @flow
 import React from 'react';
```
Comment (julianguyen) [inline client/app/components/Form/DynamicForm.jsx]: Let's do `import React, { useState } from 'react';` so we can use `useState` directly below.

File: client/app/components/Form/index.jsx [hunk 4]
```diff
@@ -1,5 +1,6 @@
 // @flow
 import React from 'react';
+import type { Node } from 'react';
```
Comment (julianguyen) [inline client/app/components/Form/index.jsx]: Similar comment as above about `useState`!

File: client/app/components/Input/utils.js [hunk 5]
```diff
@@ -57,7 +57,7 @@ export type Props = {
     | 'hidden'
     | 'switch'
     | 'location',
-  name?: string,
+  name: string,
```
Comment (julianguyen) [inline client/app/components/Input/utils.js]: Could you explain why you made this a required value?

File: INSTALLATION.md (hunk 1 above)
Comment (alekpentchev) [reply INSTALLATION.md]: I agree, this file shoudn't occur in this PR and I don't know why it did. I'll check that.

File: client/app/components/Form/DynamicForm.jsx (hunk 2 above)
Comment (alekpentchev) [reply client/app/components/Form/DynamicForm.jsx]: Yes, strange. Even though when I've reverted this change and there wasn't any error in vs-code, circleci test failed on that line exactly :/

File: client/app/components/Form/DynamicForm.jsx (hunk 3 above)
Comment (alekpentchev) [reply client/app/components/Form/DynamicForm.jsx]: Yes, sure. Missed that, thank you.

File: client/app/components/Input/utils.js (hunk 5 above)
Comment (alekpentchev) [reply client/app/components/Input/utils.js]: It was the same problem as with nameValue. Keeping this property as optional resulted in having an error in the code of either DynamicForm or Form.

File: client/app/components/Form/index.jsx (hunk 4 above)
Comment (alekpentchev) [reply client/app/components/Form/index.jsx]: Unfortunately, I cannot import type like this. See below:

image

File: client/app/components/Input/utils.js (hunk 5 above)
Comment (alekpentchev) [reply client/app/components/Input/utils.js]: If 'name' is marked as optional I get such error in the code:

image

File: INSTALLATION.md (hunk 1 above)
Comment (alekpentchev) [reply INSTALLATION.md]: @julianguyen Ok, I see why it's here. I've ifme repo in 2018 and did a change on my fork. Now when I've created this PR to ifme this files shows. So I'll create a new PR with all your comments from this PR applied. Sorry for any inconvenience.

**Comment by alekpentchev** [comment]
PR closed due to outdated fork with old and hard to revert changes. New PR will be created.

**Pull request closed without merging**